Striding for lists Part 2

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49352 Striding for lists Part 2**

Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

tugsbayasgalan committed:
Striding for lists Part 2

Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

eellison commented on torch/csrc/jit/frontend/ir_emitter.cpp:
```diff
@@ -3462,22 +3461,10 @@ struct to_ir {
-    if (!step) {
-      step = graph->insertConstant(1, loc);
-    }
-    NamedValue step_nv = NamedValue(loc, "step", step);
```
nice cleanup !!

eellison commented on torch/csrc/jit/runtime/register_ops_utils.cpp:
```diff
@@ -428,9 +428,14 @@ void listMulIntRight(Stack* stack) {
+  auto end_val = pop(stack);
+  auto start_val = pop(stack);
+
+  int64_t step = step_val.isInt() ? step_val.to<int64_t>() : 1;
```
I would add a comment for why we're setting the values to INT64_MAX if they're None, it's confusing without context.

tugsbayasgalan replied:
sounds good

eellison commented on torch/csrc/jit/runtime/register_ops_utils.cpp:
```diff
@@ -428,9 +428,14 @@ void listMulIntRight(Stack* stack) {
+  auto end_val = pop(stack);
+  auto start_val = pop(stack);
+
+  int64_t step = step_val.isInt() ? step_val.to<int64_t>() : 1;
```
why is step 1 and not INT64_MAX ? (this may be right just want to double check)

tugsbayasgalan replied:
this was initially "int step = 1" in the slice schema, but i think you are right. We should probably make it INT64_MAX to be consistent.

tugsbayasgalan replied:
When there is no step size is given, i think the python assumption is that step size is 1. Because if you do x[1:3], it outputs all elements starting from index 1 to index 3 with one increment.

eellison commented on test/test_jit.py:
```diff
@@ -4701,6 +4701,10 @@ def consec(size, start=0):
             numel = torch.tensor(size).prod().item()
             return torch.arange(numel).view(size)
 
+        # copied from @agolynski
```
you could maybe remove the `copied from ...` line, not sure that it really helps the test much

tugsbayasgalan replied:
haha ok. i thought we needed to let people know that this is a copied code

eellison reviewed (commented):
Looks great! Couple comments but it looks pretty much ready to land to me. Do we also need to update the schema for negative striding with strings ?

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

eellison commented on test/test_jit.py:
```diff
@@ -4761,6 +4764,17 @@ def func(x, value1, value2):
 
+        # Torchscript assumes type Tensor by default, so we need this explicit
+        # declaration.
+        def check_indexing_list_int(indexing, list):
```
Can we add some negative str slicing tests ?

tugsbayasgalan replied:
yep

eellison reviewed (commented):
This looks great also; I think we're just missing the negative stride str tests/logic.

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

eellison reviewed (approved):
Looks great, ship it (after waiting two weeks following PT1 ....)

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

SplitInfinity commented:
@eellison Should I tag some of the MS ONNX people on this or is that unnecessary?

tugsbayasgalan commented:
Hi @BowenBao and @neginraoof, could you guys take a look at changes I made under onnx/symbolic_opset? Since I am not too familiar with onnx, it would be nice to know if I am doing something wrong.

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Striding for lists Part 2"


Summary: In this PR, we replace all definitions of slice to take None parameters for the start, end, and step. This will simplify the compiler logic

Test Plan: test_jit test cases

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25929903](https://our.internmc.facebook.com/intern/diff/D25929903)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D25929903).
